### PR TITLE
Fix missing key prop in CameraRoll example

### PR DESCRIPTION
**Pull request by rh389**
Fixes https://github.com/facebook/react-native/issues/12132

### CODE REVIEW

File: Examples/UIExplorer/js/CameraRollView.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ var CameraRollView = React.createClass({
         var imageStyle = [styles.image, {width: imageSize, height: imageSize}];
         return (
           <Image
+            key={asset}
```
Comment (mkonicek) [inline Examples/UIExplorer/js/CameraRollView.js]: What's the type of `asset`? Is it `Image`? Is it OK to use an `Image` object as a React key? Probably best to use its path or something else?

Comment (rh389) [inline Examples/UIExplorer/js/CameraRollView.js]: It is an Image yes, I was following the lead of https://github.com/facebook/react-native/blob/master/Examples/UIExplorer/js/CameraRollExample.js. AFAIK it's fine to use any type as a key but possibly sub-optimal if object references change when their contents don't materially change. Happy to go with path instead if you like - it's probably safe to assume that's unique in this context.

Comment (mkonicek) [inline Examples/UIExplorer/js/CameraRollView.js]: Hmm if it's used in the other example I'm OK with it although didn't know that keys could be objects.

Comment (mkonicek) [inline Examples/UIExplorer/js/CameraRollView.js]: From React docs: "A key is a string": https://facebook.github.io/react/docs/lists-and-keys.html

Could you use the path please just to make this 100% legit?

The other example might be using the key wrong.

**Comment by mkonicek** [comment]
Sorry about the long delay reviewing this and thank you for the fix!

**Comment by rh389** [comment]
Hi @mkonicek - thanks for the review. I've updated both examples of `key` to use `asset.node.image.uri` instead, which is a string like `assets-library://asset/asset.JPG?id=99D53A1F-FEEF-40E1-8BB3-7DD55A43C8B7&ext=JPG`. As far as I can tell it's always-present and unique on image assets. Also flow-typed `asset`.

(Glad to learn something new about keys too! Makes sense, and I although think objects might accidentally work in some cases, I'll now know to stick to strings).

File: Examples/UIExplorer/js/CameraRollView.js [hunk 2]
```diff
@@ -92,11 +92,12 @@ var CameraRollView = React.createClass({
       batchSize: 5,
       imagesPerRow: 1,
       assetType: 'Photos',
-      renderImage: function(asset) {
+      renderImage: function(asset: Image) {
```
Comment (shergin) [inline Examples/UIExplorer/js/CameraRollView.js]: `Image` as a type? 🤔
How does it work?

Comment (rh389) [inline Examples/UIExplorer/js/CameraRollView.js]: This isn't my design but that's what they are - the `CameraRollView` is a `ListView` where each row's data is an array of `Image`s (as in instances of the `Image` RN component). See for example [`getInitialState`](https://github.com/facebook/react-native/blob/44317996f361db4347c626be1e1499acba58c281/Examples/UIExplorer/js/CameraRollView.js#L113) and [`_rowHasChanged`](https://github.com/facebook/react-native/blob/44317996f361db4347c626be1e1499acba58c281/Examples/UIExplorer/js/CameraRollView.js#L209). Apparently that's just used as a container for data from which the rendered `Image` is built.

**Pull request closed without merging**